frontend/src/components/pages/shadowlinks/create/connection/mtls-configuration.test.tsx

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form } from 'components/redpanda-ui/components/form';
+import { useForm } from 'react-hook-form';
+import { fireEvent, render, screen, waitFor } from 'test-utils';
+
+import { MtlsConfiguration } from './mtls-configuration';
+import { FormSchema, type FormValues, initialValues, TLS_MODE } from '../model';
+
+const TestWrapper = ({ defaultValues = initialValues }: { defaultValues?: FormValues }) => {
+  const form = useForm<FormValues>({
+    resolver: zodResolver(FormSchema),
+    defaultValues,
+    mode: 'onBlur',
+  });
+
+  const onSubmit = () => {
+    // Test form submission
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)}>
+        <MtlsConfiguration />
+        <button data-testid="submit-button" type="submit">
+          Submit
+        </button>
+      </form>
+    </Form>
+  );
+};
+
+describe('MtlsConfiguration', () => {
```

Comment:
Controversial take: only test side effects, action user can take, such as configuring mTLS, uploading a file with certificate, rendering empty state perhaps. Do not test things such as rendering each field properly, because:
- unit tests would cover the field transformation such as timestamp formatting, size unit parsing etc.
- we have a proto/connect contract in place so we can guarantee the UI will just render whatever the backend returns, we no longer need to maintain a mapping layer since it's just based on what proto tells us
- if a test fails, that should give us confidence there is a real failure, but right now I feel like the only useful test here that would stop me from merging is the one where I verify mTLS uploads via file or direct text.